### ISSUE TITLE
Fix S3254 FN: primary constructors

### ISSUE THREAD

**Issue by antonioaversa** [SonarSource/sonar-dotnet#8096]
[S3254](https://sonarsource.github.io/rspec/#/rspec/S3254) should raise when default values are passed to parameters of [primary constructors](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-12#primary-constructors).

```cs
_ = new C1(42);     // FN
_ = new C1(41 + 1); // FN, expression results into the default value

class C1(int i = 42);
```

.. and also on
* object creation initializer
* PrimaryConstructorBaseInitiliazer
